src/Essential.Diagnostics.RollingFileTraceListener/IO/FileSystem.cs

```diff
@@ -19,6 +26,13 @@ public class FileSystem : IFileSystem
         /// <returns></returns>
         public Stream Open(string path, FileMode mode, FileAccess access, FileShare share)
         {
+            if (CreateSubdirectories)
+            {
+                // Making sure that all subdirectories in file path exists
+                var directory = Path.GetDirectoryName(path);
```

Comment:
The IFileSystem interface should be a wrapper around the IO static methods, so they can be substituted for testing.

i.e. Add methods to the interface for GetDirectoryName() and CreateDirectory(), each of which forwards directly to the corresponding static method.

e.g.
```
        public DirectoryInfo CreateDirectory(string path)
        {
            return Directory.Create(path);
        }
```

Do the same for GetDirectoryName(), and add doc comments (copy them from the .NET Framework documentation).

Keep the wrapper very simple, because it can't be tested, the IFileSystem is replaced in unit tests.

Business logic, such as "if (flag) { get directory name, create directory }" would then be in the RollingTextWriter (and can be tested).